scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/context/TaskContextVariableExtractor.java

```diff
@@ -122,11 +122,17 @@
 
     public Map<String, Serializable> extractScopeVariables(TaskContext taskContext) {
         Map<String, Serializable> variables = new HashMap<>();
-
+        Map<String, Serializable> previousVariables = null;
+        try {
+            previousVariables = extractVariables(taskContext, false);
+        } catch (IOException | ClassNotFoundException e) {
+            logger.error("Unable to extract variables", e);
```

Comment:
Is the result correct if we cannot get the previous variables (I think it is not)? I think you should notify the caller through an exception.

Reply:
Yes, i have just fixed that, as also suggested by Marco above :)